mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,623 @@
+        `ndim == 1`, it is assumed to be event indices, and all events will be
+        grouped together.
+    tmin : float | list of floats, shape (n_pac_windows,) | None
+        If `events` is not provided, it is the start time to use in `inst`.
+        If `events` is provided, it is the time (in seconds) to include before
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `min(inst.times)`.
+    tmax : float | list of floats, shape (n_pac_windows,) | None
+        If `events` is not provided, it is the stop time to use in `inst`.
+        If `events` is provided, it is the time (in seconds) to include after
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `max(inst.n_times)`.
+    n_cycles : int | None
+        The number of cycles to be included in the window for each band-pass
+        filter. Defaults to 3.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., `sklearn.preprocessing.scale`.
+        Defaults to no scaling.
+    return_data : bool
+        If False, output will be `[pac_out]`. If True, output will be,
+        `[pac_out, low_freq_phase, hi_freq_amplitude]`.
+    concat_epochs : bool
+        If True, epochs will be concatenated before calculating PAC values. If
+        epochs are relatively short, this is a good idea in order to improve
+        stability of the PAC metric.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see `mne.verbose`).
+
+    Returns
+    -------
+    pac_out : array, list of arrays, dtype float, shape (n_epochs, n_pairs,
+              n_times)
+        The computed phase-amplitude coupling between
+        each pair of data sources given in ixs. If multiple pac metrics are
+        specified, there will be one array per metric in the output list.
+    [phase_signal] : array, shape (n_phase_signals, n_times,)
+        Only returned if `return_data` is True. The phase timeseries of the
+        phase signals (first column of `ixs`).
+    [amp_signal] : array, shape (n_amp_signals, n_times,)
+        Only returned if `return_data` is True. The amplitude timeseries of the
+        amplitude signals (second column of `ixs`).
+
+    References
+    ----------
+    [1] This function uses the PacPy module developed by the Voytek lab.
+        https://github.com/voytekresearch/pacpy
+    """
+    from mne.io.base import _BaseRaw
+    if not isinstance(inst, _BaseRaw):
+        raise ValueError('Must supply Raw as input')
+    sfreq = inst.info['sfreq']
+    data = inst[:][0]
+    pac = _phase_amplitude_coupling(data, sfreq, f_phase, f_amp, ixs,
```

Comment:
I usually assign things to variables for docstring purposes. If folks think it's fine to just return the output of the hidden function then we can do that.

Also, piggy backing off of @agramfort's comments, you two are imagining basically that the `_phase_amplitude_coupling` function (and supporting functions) would go into pacpy? And then the public function would be the only contribution to MNE-sandbox? Looking through the code it might not be too bad, since I wrote the hidden function in order to avoid dependencies on MNE objects...